### PR TITLE
gh-123370: Fix the canvas not clearing after running turtledemo.clock

### PR DESCRIPTION
**Pull request by Wulian233**
In addition to fixing this bug, I also made some changes. 
1. The second hand was shorter than the minute hand, which was incorrect, and this has now been fixed. (minute hand 130 → 115)
2. Remove the first line's `coding: cp1252`, as this file has always been utf8. 

* Issue: gh-123370

### CODE REVIEW

File: Lib/turtledemo/clock.py [hunk 1]
```diff
@@ -52,62 +51,54 @@ def clockface(radius):
             jump(-radius)
         rt(6)
 
+def display_date_time():
+    writer.clear()
+    now = datetime.now()
+    writer.home()
+    writer.forward(distance=65)
+    writer.write(now.strftime(format="%A"),
+                 align="center", font=("TkFixedFont", 14, "bold"))
```
Comment (terryjreedy) [inline Lib/turtledemo/clock.py]: After adding "dtfont = "TkFixedFont", 14, "bold" above (perhaps after d-t import),

```suggestion
    writer.write(now.strftime(format="%A"), align="center", font=dtfont)
```
and ditto for next write.

File: Lib/turtledemo/clock.py [hunk 2]
```diff
@@ -52,62 +51,54 @@ def clockface(radius):
...
-    return wochentag[t.weekday()]
-
-def datum(z):
-    monat = ["Jan.", "Feb.", "Mar.", "Apr.", "May", "June",
-             "July", "Aug.", "Sep.", "Oct.", "Nov.", "Dec."]
-    j = z.year
-    m = monat[z.month - 1]
-    t = z.day
-    return "%s %d %d" % (m, t, j)
+    display_date_time()
```
Comment (terryjreedy) [inline Lib/turtledemo/clock.py:94]: Why is this here instead of in tick?  Update is rare, but not impossible.  Does the rewriting lock the text?

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Nice.  Needs one change and one question answered.  I also need to verify that we can remove functions. I suggested other changes on issue for other PR/issue.

File: Lib/turtledemo/clock.py [hunk 3]
```diff
@@ -12,29 +11,29 @@
...
-    right(winkel)
+    right(angle)
     forward(distanz)
-    left(winkel)
+    left(angle)
     pendown()
 
-def hand(laenge, spitze):
-    fd(laenge*1.15)
+def hand(spitze, tip):
```
Comment (terryjreedy) [inline Lib/turtledemo/clock.py]: I meant change 'laenge' to 'length' and 'spitze' to 'tip'.  (and 'distanz' to 'distance' and 'winkel' to 'angle').  Please revert and don't do anything with names until I verify that we can change parameter names in this file.

File: Lib/turtledemo/clock.py (hunk 2 above)
Comment (Wulian233) [reply Lib/turtledemo/clock.py:94]: https://github.com/user-attachments/assets/9353818c-9504-4009-94a2-f60b13f29bbd

Refreshes per second will make the text look like this

The text should be static now, and if someone opens the program after midnight or adjust the device time, the date not change. We check if the day a second ago is the same as the day now, in the tick function, and then the date can be synchronized. What do you think

The bug I found was that the text kept refreshing continuously, and even after "stop", it continued to run

**Comment by Wulian233** [comment]
I have made the requested changes; please review again @terryjreedy 

The time is now dynamically updated, and the date is refreshed only once when the day changes. The cause of this bug is that the text was being refreshed every second 🚀 🕙 

https://github.com/user-attachments/assets/a2dcbe0a-f718-4be6-9c98-789451d9c6f7

File: Misc/NEWS.d/next/Library/2024-08-28-19-27-35.gh-issue-123370.SPZ9Ux.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Fix the canvas not clearing after running turtledemo clock, and improve the
+clock example suite
```
Comment (terryjreedy) [inline Misc/NEWS.d/next/Library/2024-08-28-19-27-35.gh-issue-123370.SPZ9Ux.rst]:
```suggestion
Fix the canvas not clearing after running turtledemo clock.
```

File: Lib/turtledemo/clock.py [hunk 5]
```diff
@@ -52,65 +54,60 @@ def clockface(radius):
...
+    hour_hand = initialize_hand("hour_hand", ("blue3", "red3"))
+
     ht()
     writer = Turtle()
-    #writer.mode("logo")
     writer.ht()
     writer.pu()
     writer.bk(85)
-
-def wochentag(t):
```
Comment (terryjreedy) [inline Lib/turtledemo/clock.py:82]: Backports should not break code.  While someone importing and calling these two functions would be unexpected, it is not impossible (EDITED), and other coredevs agreed that we should follow the rule even here.  So restore these two functions for this fix PR, which can then be backported.  We can (I believe), skip the deprecation process in a main-only followup that deletes these and replaces German.

**Review by terryjreedy — CHANGES_REQUESTED** [review]

**Comment by terryjreedy** [comment]
I cannot review until my Python development machine is repaired or I add everything to another machine.

**Comment by Wulian233** [comment]
I understand, and there's no rush🙂

**Comment by Wulian233** [comment]
It's been almost 2 months and there is a duplicate report https://github.com/python/cpython/issues/125187 for this bug, do I need to do anything else with this PR @terryjreedy 

I plan to slove these issues related to https://github.com/python/cpython/issues/123370#issuecomment-2322733087 after this PR merge🙂

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Clock looks fine.  Everything disappears either on STOP or selection of another example.  The one change needed is to revert all changes to wochentag and datum.  Since they are only left for back-compatibility, we do *not* want to change behavior.  In any case, changing them has nothing to do with fixing the bug.

What I think we should do is move them down just above def main with an added comment `# Unused; kept for back compatibility.` on the header lines. The new comment is needed because of the bugfix and hence relevant to the PR.

I thought of asking for implementation of my new item 5 issue comment (use English local names) but the same 'not needed here' comment applies.

**Comment by Wulian233** [comment]
My idea is to minimize the changes to this PR and only fix the bug. revert all changes to wochentag and datum. Later use English local names, PR called the modern turtledemo, while including all your suggetions and the new changes brought about by this pr (ISO format time) , but the bug that the minute hand is longer than the second hand can be retained, respecting reality :)

I have made the requested changes; please review again

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
You reverted more than I requested, but moving the other changes, which I do want, to a separate cleanup/modernize backported PR is fine.

**Comment by terryjreedy** [comment]
Merge conflict explained at https://github.com/python/cpython/issues/118673#issuecomment-2419980443.  This will import docstring changes in rest of turtledemo files.  I have fixed this one for 3.13 and will backport that PR.
